packages/xdl/src/Project.ts

```diff
@@ -892,26 +902,87 @@ export async function publishAsync(
         'generated',
         'AppConstants.java'
       );
-      await ExponentTools.deleteLinesInFileAsync(
-        `START EMBEDDED RESPONSES`,
-        `END EMBEDDED RESPONSES`,
-        constantsPath
-      );
-      await ExponentTools.regexFileAsync(
-        '// ADD EMBEDDED RESPONSES HERE',
-        `
-        // ADD EMBEDDED RESPONSES HERE
-        // START EMBEDDED RESPONSES
-        embeddedResponses.add(new Constants.EmbeddedResponse("${fullManifestUrl}", "assets://shell-app-manifest.json", "application/json"));
-        embeddedResponses.add(new Constants.EmbeddedResponse("${androidManifest.bundleUrl}", "assets://shell-app.bundle", "application/javascript"));
-        // END EMBEDDED RESPONSES`,
-        constantsPath
-      );
-      await ExponentTools.regexFileAsync(
-        /RELEASE_CHANNEL = "[^"]*"/,
-        `RELEASE_CHANNEL = "${options.releaseChannel}"`,
-        constantsPath
-      );
+      if (fs.existsSync(constantsPath)) {
+        // This is an ExpoKit app
+        // We need to add EmbeddedResponse instances on Android to tell the runtime
+        // that the shell app manifest and bundle is packaged.
+        await ExponentTools.deleteLinesInFileAsync(
+          `START EMBEDDED RESPONSES`,
+          `END EMBEDDED RESPONSES`,
+          constantsPath
+        );
+        await ExponentTools.regexFileAsync(
+          '// ADD EMBEDDED RESPONSES HERE',
+          `
+          // ADD EMBEDDED RESPONSES HERE
+          // START EMBEDDED RESPONSES
+          embeddedResponses.add(new Constants.EmbeddedResponse("${fullManifestUrl}", "assets://shell-app-manifest.json", "application/json"));
+          embeddedResponses.add(new Constants.EmbeddedResponse("${androidManifest.bundleUrl}", "assets://shell-app.bundle", "application/javascript"));
+          // END EMBEDDED RESPONSES`,
+          constantsPath
+        );
+        await ExponentTools.regexFileAsync(
+          /RELEASE_CHANNEL = "[^"]*"/,
+          `RELEASE_CHANNEL = "${options.releaseChannel}"`,
+          constantsPath
+        );
+      } else {
+        // This is an app with expo-updates installed, so set the applicable meta-data properties in
+        // AndroidManifest.xml
```

Comment:
As far as I know, managed apps don't have `AndroidManifest.xml` (an `android` folder) and I haven't been aware that `expo publish` might be used with bare apps.

The `if` branch above says it's for ExpoKit apps. Then what kind of apps can land in this `else` branch? Additionally, what does the workflow enabled by this code look like for the user?

Reply:
This `else` branch is for bare workflow apps with `expo updates`, in which case `expo publish` will need to be used in order to host an update on Expo's servers. The workflow is similar to how OTA updates with ExpoKit apps work right now, but I expect to improve it over the coming months after pushing out this initial version with SDK 37.

Reply:
I've changed this to just check for `expo-updates` instead of using this `else` branch -- this should be more clear. (thanks for the suggestion @sjchmiela !)